packages/datastore/__tests__/conflictResolutionBehavior.test.ts

```diff
@@ -674,178 +265,97 @@ describe('DataStore sync engine', () => {
 						]);
 
-						expectFinalRecordsToMatch({
-							postId: original.id,
-							version: 4,
+						expect(await postHarness.currentContents).toMatchObject({
+							_version: 4,
 							title: 'update from second client',
 						});
 					});
 					test('rapid mutations on fast connection when initial create is not pending', async () => {
-						const original = await DataStore.save(
-							new Post({
-								title: 'original title',
-								blogId: 'blog id',
-							})
-						);
-
-						await waitForEmptyOutbox();
-
-						//region perform consecutive updates
-						await pause(200);
-						await revPost(original.id, 'post title 0');
+						const postHarness = await harness.createPostHarness({
+							title: 'original title',
+							blogId: 'blog id',
+						});
 
-						await pause(200);
-						await revPost(original.id, 'post title 1');
+						harness.userInputLatency = 'slowerThanOutbox';
+						harness.latency = 'low';
 
-						await externalPostUpdate({
-							originalPostId: original.id,
+						await postHarness.revise('post title 0');
+						await postHarness.revise('post title 1');
+						await harness.externalPostUpdate({
+							originalPostId: postHarness.original.id,
 							updatedFields: { title: 'update from second client' },
 							version: 1,
 						});
+						await postHarness.revise('post title 2');
 
-						await pause(200);
-						await revPost(original.id, 'post title 2');
-						//endregion
-
-						await waitForEmptyOutbox();
+						await harness.outboxSettled();
+						await harness.expectGraphqlSettledWithUpdateCallCount(4);
 
-						await graphqlServiceSettled({
-							graphqlService,
-							expectedNumberOfUpdates: numberOfUpdates,
-							externalNumberOfUpdates: 1,
-							modelName: 'Post',
-						});
-
-						expect(subscriptionLog).toEqual([
+						expect(harness.subscriptionLogs()).toEqual([
 							['original title', 1],
 							['post title 0', 1],
 							['post title 1', 1],
 							['post title 2', 1],
 							['post title 0', 5],
 						]);
 
-						expectFinalRecordsToMatch({
-							postId: original.id,
-							version: 5,
+						expect(await postHarness.currentContents).toMatchObject({
+							_version: 5,
 							title: 'post title 0',
 						});
 					});
 					test('observe on poor connection with awaited outbox', async () => {
-						const original = await DataStore.save(
-							new Post({
-								title: 'original title',
-								blogId: 'blog id',
-							})
-						);
-
-						await waitForEmptyOutbox();
-
-						graphqlService.setLatencies({
-							request: latency,
-							response: latency,
-							subscriber: latency,
-							jitter,
+						const postHarness = await harness.createPostHarness({
+							title: 'original title',
+							blogId: 'blog id',
 						});
 
-						//region perform consecutive updates
-						await revPost(original.id, 'post title 0');
-						await waitForEmptyOutbox();
+						harness.userInputLatency = 'slowerThanOutbox';
```

Comment:
We don't pause between revisions on the original test - is there a reason to do that here?

```suggestion
						harness.userInputLatency = 'fasterThanOutbox';
```